util/Map.h

```diff
@@ -63,32 +88,18 @@
     #endif
 #endif
 
-struct Map_CONTEXT
-{
-    #ifdef Map_ENABLE_KEYS
-        uint32_t* hashCodes;
-        uint32_t* hashIndexes;
-        Map_KEY_TYPE* keys;
-    #endif
-
-    #ifdef Map_ENABLE_HANDLES
-        uint32_t* handles;
-        uint32_t nextHandle;
-    #endif
-
-    Map_VALUE_TYPE* values;
-
-    uint32_t count;
-    uint32_t capacity;
-
-    struct Allocator* allocator;
-};
-
 static inline struct Map_CONTEXT* Map_FUNCTION(new)(struct Allocator* allocator)
 {
-    return Allocator_clone(allocator, (&(struct Map_CONTEXT) {
+    struct Map_CONTEXT* map = Allocator_clone(allocator, (&(struct Map_CONTEXT) {
         .allocator = allocator
     }));
+    #ifdef Map_ENABLE_KEYS
+        struct Allocator* tempAlloc = Allocator_child(allocator);
+        struct Random* rand = Random_new(tempAlloc, NULL, NULL);
```

Comment:
In fact, siphash only need a unique random 16 bytes secret key for each map during its life circle, so once code get the 16 bytes random data, I released the random object.